python/ray/serve/backend_state.py

```diff
@@ -272,8 +283,7 @@ def cleanup(self):
             return
 
         try:
-            ray.util.remove_placement_group(
-                ray.util.get_placement_group(self._placement_group_name))
+            ray.util.remove_placement_group(self._placement_group)
```

Comment:
hmm do we need to check that we've set `self._placement_group` here? seems like there are cases where it might be `None`

Reply:
i think our current implementation didn't guard against None placement group either, which will throw AssertionError that we didn't catch. I can just add a simple check then